checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -492,21 +751,32 @@ private boolean shouldTrackInvocationResult(Set<Obligation> obligations, Node no
 
       if (TreeUtils.isSuperConstructorCall(methodInvokeTree)
           || TreeUtils.isThisConstructorCall(methodInvokeTree)) {
-        handleThisOrSuperConstructorMustCallAlias(obligations, node);
+        Node mustCallAliasArgument = getMustCallAliasArgumentNode(node);
+        // If the MustCallAlias argument is also in the set of Obligations, then remove it -- its
+        // must-call obligation has been fulfilled by being passed on to the MustCallAlias
+        // constructor
+        // (because a this/super constructor call can only occur in the body of another
+        // constructor).
+        if (mustCallAliasArgument instanceof LocalVariableNode) {
+          removeObligationsContainingVar(obligations, (LocalVariableNode) mustCallAliasArgument);
+        }
         return false;
       }
       return !returnTypeIsMustCallAliasWithUntrackable((MethodInvocationNode) node)
           && !hasNotOwningReturnType((MethodInvocationNode) node);
     }
+    // Constructor results from new expressions are always owning.
     return true;
   }
 
   /**
    * Returns true if this node represents a method invocation of a must-call-alias method, where the
-   * other must call alias is untrackable: an owning field or a pointer that is guaranteed to be
-   * non-owning, such as {@code "this"} or a non-owning field. Owning fields are handled by the rest
-   * of the checker, not by this algorithm, so they are "untrackable". Non-owning fields and this
-   * nodes are guaranteed to be non-owning, and therefore do not need to be tracked, either.
+   * argument in the must-call-alias position is untrackable: an owning field or a pointer that is
```

Comment:
This gives a good operational definition of "untrackable", but please also state its meaning.  Does it mean that the expression is definitely non-owning, or something else?

Reply:
It means what it says: that the algorithm isn't going to track the given expression. Non-owning expressions are untrackable, but as this says so are e.g. owning fields, because they're handled separately.

It's clear that "untrackable" isn't sufficient to get across the point that there are certain kinds of expressions that this algorithm (i.e. the consistency checker) is not responsible for, and that this method's job is to filter those expressions out. Do you have a better suggestion? I'm concerned that just saying "untrackable" again in this comment in different words will only make the comment more verbose without adding any more content.